src/arithmetic/algebraic_expression/algebraic_expression.c

```diff
@@ -352,7 +352,10 @@ AlgebraicExpression *AlgebraicExpression_RemoveSource
 	AlgebraicExpression *prev = *root;
 	AlgebraicExpression *current = *root;
 
-	while(current->type == AL_OPERATION) {
+	/* This loop iterates until it reaches an operand node or a transpose node
+	 * with an operand child. */
+	while(current->type == AL_OPERATION &&
+		  !(current->operation.op == AL_EXP_TRANSPOSE && FIRST_CHILD(current)->type == AL_OPERAND)) {
```

Comment:
Can you explain why this loop condition was changed?
I would really like to keep the original version as it is straightforward  and easy to reason about